core/include/moveit/task_constructor/stages/pick.h

```diff
@@ -48,26 +48,30 @@ MOVEIT_CLASS_FORWARD(CartesianPath)
 
 namespace stages {
 
-/** Pick wraps a complete pipeline to pick up an object with a given end effector.
+/** PickPlaceBase wraps the pipeline to pick or place an object with a given end effector.
  *
  * Picking consist of the following sub stages:
- * - reaching to the object + "pre-grasp" end effector posture
  * - linearly approaching the object along an approach direction/twist
  * - "grasp" end effector posture
  * - attach object
  * - lift along along a given direction/twist
  *
+ * Placing consist of the inverse order of stages:
+ * - place down along a given direction
+ * - detach the object
+ * - linearly retract end effector
+ *
  * The end effector postures corresponding to pre-grasp and grasp as well as
  * the end effector's Cartesian pose needs to be provided by an external grasp stage.
  */
-class Pick : public SerialContainer {
+class PickPlaceBase : public SerialContainer {
```

Comment:
Yes, I like to avoid redundancy... I'm not sure what you propose here: Removing Pick + Place completely (enforcing the user to manually compose their task from basic stages only) or removing PickPlaceBase and implementing Pick + Place as completely independent classes. The latter doesn't really reduces complexity.

Reply:
As I said in the last sentence, this is a statement of opinion, no concrete proposal to change something..
From a user point of view for transparency(as in introspection) of the system I think it would be better to implement all combined Containers independently to support people reading the code.

The complexity of `Tasks` in MTC is already quite high to begin with, even if you only have to worry about control flow.